lighthouse-core/test/gather/driver/wait-for-condition-test.js

```diff
@@ -35,6 +35,23 @@ function createMockWaitForFn() {
   });
 }
 
+function createMockMultipleInvocationWaitForFn() {
+  /** @type {Array<{arguments: Array<*>, mockResolve(): void, mockReject(): void}>} */
+  const calls = [];
+  const mockCancelFn = jest.fn();
+  const mockFn = jest.fn().mockImplementation((...args) => {
+    const {promise, resolve, reject} = createDecomposedPromise();
+    calls.push({
+      arguments: args,
+      mockResolve: () => resolve(),
+      mockReject: () => reject(new Error('Rejected')),
+    });
+    return {promise, cancel: mockCancelFn};
+  });
+
+  return Object.assign(mockFn, {waitForCalls: calls});
```

Comment:
👀

Reply:
if these eyes are for preferring `{...` that won't work for prototype/class properties of the jest mock fn :)

Reply:
ah ya I know. I was just eying the rare case where my blanket "use spread!" doesn't work :)